src/fields_dump.cpp

```diff
@@ -189,6 +189,10 @@ void fields::load_fields_chunk_field(h5file *h5f, bool single_parallel_file,
           } else {
             if (n != ntot)
               meep::abort("grid size mismatch %zd vs %zd in fields::load", n, ntot);
+            if (*f) {
+              delete[] * f;
+              *f = NULL;
+            }
```

Comment:
```suggestion
            delete[] *f;
```
According to the C++ standard, `delete[]` should be safe (do nothing) for a `NULL` pointer.  And there's no point in assigning it to `NULL` since we immediately re-assign `*f` on the next line.

Similarly for the other part of this PR.